### DISCUSSION TITLE
unable to connect to DB

### DISCUSSION THREAD

**Discussion by adevcoder** [dani-garcia/vaultwarden#4006] [Q&A]
I am using portainer and pulling the image vaultwarden/server:latest

the /config and /data is attached to a nfs volume 

the error i  am getting is 
```
[vaultwarden::util][WARN] Can't connect to database, retrying: Failed to create pool.
[CAUSE] Error(

,
)
```

I've tried restarting and creating a new instance and it still gives me that error when starting.

![Xnip2023-10-27_00-51-26](https://github.com/dani-garcia/vaultwarden/assets/50393117/35562355-abc0-4bab-a5b7-80f12268046e)

Update:
Added image for reference

**Comment by Masgalor** [comment]
What type of database do you use, am I assuming correctly that it is sqlite?

Sqlite does not work well with nfs shares, the official faq says:

>But use caution: this locking mechanism might not work correctly if the database file is kept on an NFS filesystem. This is because fcntl() file locking is broken on many NFS implementations. You should avoid putting SQLite database files on NFS if multiple processes might try to access the file at the same time.

[Source](https://www.sqlite.org/faq.html#q5)

**Comment by renekuehl** [comment]
Hm, got the same Error with SQLite DB on NFS as of v1.3.0.. Will test moving the Database off the NFS.. Wanted to do it a long Time ago, but it worked ;)

**Reply by BlackDex** [reply]
SQLite and NFS do not work well together.
Because of the locking which isn't atomic.
